pykokkos/lib/ufuncs.py

```diff
@@ -3077,3 +3077,390 @@ def isfinite(view):
                             view=view,
                             out=out)
     return out
+
+@pk.workunit
+def round_impl_1d_double(tid: int, view: pk.View1D[pk.double], out: pk.View1D[pk.double]):
+    out[tid] = round(view[tid])
+
+
+@pk.workunit
+def round_impl_2d_double(tid: int, view: pk.View2D[pk.double], out: pk.View2D[pk.double]):
+    for i in range(view.extent(1)):
+        out[tid][i] = round(view[tid][i])
+
+
+@pk.workunit
+def round_impl_3d_double(tid: int, view: pk.View3D[pk.double], out: pk.View3D[pk.double]):
+    for i in range(view.extent(1)):
+        for j in range(view.extent(2)):
+            out[tid][i][j] = round(view[tid][i][j])
+
+@pk.workunit
+def round_impl_1d_float(tid: int, view: pk.View1D[pk.float], out: pk.View1D[pk.float]):
+    out[tid] = round(view[tid])
+
+
+@pk.workunit
+def round_impl_2d_float(tid: int, view: pk.View2D[pk.float], out: pk.View2D[pk.float]):
+    for i in range(view.extent(1)):
+        out[tid][i] = round(view[tid][i])
+
+
+@pk.workunit
+def round_impl_3d_float(tid: int, view: pk.View3D[pk.float], out: pk.View3D[pk.float]):
+    for i in range(view.extent(1)):
+        for j in range(view.extent(2)):
+            out[tid][i][j] = round(view[tid][i][j])
+
+def round(view):
+    """
+    Rounds each element of the input view to the nearest integer-valued number.
+
+    Parameters
+    ----------
+    view : pykokkos view
+           Should have a numeric data type.
+
+    Returns
+    -------
+    out: pykokkos view
+         A view containing the rounded result for each element in
+         the input view. The returned view must have the same data
+         type as the input view.
+
+    Notes
+    -----
+    If view element ``i`` is already integer-valued, the result is ``i``.
+
+    """
+    if "int" in str(view.dtype):
+        # special case defined in API std
+        return view
+    out = pk.View(view.shape, dtype=view.dtype)
+    if len(view.shape) > 3:
+        raise NotImplementedError("only up to 3D views currently supported for round() ufunc.")
+        
+    if "double" in str(view.dtype):
+        if view.shape == ():
+            pk.parallel_for(1, round_impl_1d_double, view=view, out=out)
+        elif len(view.shape) == 1:
+            pk.parallel_for(view.shape[0], round_impl_1d_double, view=view, out=out)
+        elif len(view.shape) == 2:
+            pk.parallel_for(view.shape[0], round_impl_2d_double, view=view, out=out)
+        elif len(view.shape) == 3:
+            pk.parallel_for(view.shape[0], round_impl_3d_double, view=view, out=out)
+
+    elif "float64" in str(view.dtype):
```

Comment:
instead of having `str(view.dtype)` each time, can we save this into a variable and use it when needed